src/compat/posix/util/time.c

```diff
@@ -9,27 +9,55 @@
  *         - clock()
  */
 
+#include <errno.h>
 #include <time.h>
 #include <hal/clock.h>
 #include <kernel/time/clock_source.h>
 #include <kernel/time/ktime.h>
+#include <kernel/time/time.h>
 
 clock_t clock(void) {
 	return clock_sys_ticks();
 }
 
 int clock_gettime(clockid_t clk_id, struct timespec *ts) {
-	ktime_get_timespec(ts);
+	switch(clk_id) {
+		case CLOCK_REALTIME: {
+			getnsofday(ts, NULL);
+			break;
+		}
+		case CLOCK_MONOTONIC: {
+			ktime_get_timespec(ts);
+			break;
+		}
```

Comment:
You do not need `{` in cases. Why you use them? It's convenient or what? (as for me - not)

Reply:
Thanks. Done